src/platform/Darwin/DnssdImpl.h

```diff
@@ -48,6 +49,7 @@ struct GenericContext
 };
 
 struct RegisterContext;
+struct RegisterRecordContext;
```

Comment:
I *think* that this is not needed. The reason why `RegisterContext` is forward declared is because it is used in `CHIP_ERROR MdnsContexts::GetRegisterContextOfType(const char * type, RegisterContext ** context);` but nothing uses `RegisterRecordContext` before it is fully declared.

Reply:
done. copy paste :D